napari/components/add_layers_mixin.py

```diff
@@ -849,40 +850,28 @@ def _add_layers_with_plugins(
         """
         layer_data = read_data_with_plugins(path_or_paths, plugin=plugin)
 
-        if not layer_data:
-            # if layer_data is empty, it means no plugin could read path
-            # we just want to provide some useful feedback, which includes
-            # whether or not paths were passed to plugins as a list.
-            if isinstance(path_or_paths, (tuple, list)):
-                path_repr = f"[{path_or_paths[0]}, ...] as stack"
+        # glean layer names from filename. These will be used as *fallback*
+        # names, if the plugin does not return a name kwarg in their meta dict.
+        if isinstance(path_or_paths, str):
+            filenames = itertools.repeat(path_or_paths)
+        elif is_sequence(path_or_paths):
+            if len(path_or_paths) == len(layer_data):
+                filenames = iter(path_or_paths)
             else:
-                path_repr = path_or_paths
-            msg = f'No plugin found capable of reading {path_repr}.'
-            logger.error(msg)
-            return []
+                # hard to say what to do if a list of paths have been returned
+                # as a list of layer data without a 1:1 relationship.
+                # here we just use the first name...
```

Comment:
```suggestion
                # if a list of paths has been returned as a list of layer data
                # without a 1:1 relationship between the two lists we just use
                # iterate over the first name
```
Took me a while to parse the above - i think this comment is now more clear, i agree it's an odd case

Reply:
yup I agree, that's clearer